### ISSUE TITLE
Allowing CatBoost for `model_tau` input to Explainer class

### ISSUE THREAD

**Issue by portchester1989** [uber/causalml#826]
**Is your feature request related to a problem? Please describe.**
Even if I would like to use CatBoost for `model_tau` for `Explainer` class, I cannot bc there's not exposure of `feature_importances_`. And in case we have 

**Describe the solution you'd like**
modify `check_condition` and `default_importance` methods under `Explainer` class

**Describe alternatives you've considered**
custom implementation of Explainer class on my end

**Additional context**

**Comment by aman-coder03** [comment]
hey @jeongyoonlee 
i have reviewed `explainer.py` in full
the fix touches 2 methods plus a small private helper.

## Root cause
`check_conditions` validates the model with...
```python
model_test.fit([[0], [1]], [0, 1])
assert hasattr(model_test, "feature_importances_")
```

CatBoost exposes `get_feature_importance()` instead of `feature_importances_`, so this assertion raises. `default_importance` then hardcodes the same attribute access, which would also fail even if the first check were bypassed

## Planned changes
add a private helper that normalises importance access across sklearn/LightGBM/XGBoost and CatBoost...
```python
def _get_feature_importances(model):
    if hasattr(model, "feature_importances_"):
        return model.feature_importances_
    elif hasattr(model, "get_feature_importance"):
        return model.get_feature_importance()
    raise AttributeError(
        "model_tau must expose feature importances via `feature_importances_` "
        "or `get_feature_importance()` (after fitting)."
    )
```

then replacing the `hasattr` assertion in `check_conditions` and the hardcoded attribute access in `default_importance` with calls to this helper. Will also update the `model_tau` docstring to include CatBoost

## What is not changing
`perm_importance` already works with any estimator, and CatBoost SHAP support is a separate concern, so both are out of scope.
No public API changes

## Tests
i will add a test with `CatBoostRegressor` as `model_tau` and `method="auto"`, verifying it completes without error and returns a correctly shaped result

**Comment by jeongyoonlee** [comment]
Thanks for tracing this through — your plan is exactly right. A small private helper that normalizes importance access (`feature_importances_` for sklearn/LightGBM/XGBoost vs. `get_feature_importance()` for CatBoost), used in both `check_conditions()` (`explainer.py:82`, the `hasattr` assert) and `default_importance()` (`explainer.py:152`), is the correct, minimal fix. Keeping `perm_importance` and CatBoost SHAP out of scope is the right call.

Two things to confirm while implementing:
- The dummy-fit in `check_conditions()` must actually fit CatBoost before importances are available — verify the helper works against a *fitted* CatBoost, not just attribute presence.
- CatBoost is an optional dependency, so guard the new test with `pytest.importorskip("catboost")` so core CI (which doesn't install CatBoost) stays green.

Please also add the `CatBoostRegressor` + `method="auto"` test you described. Go ahead.
